contracts/src/v0.8/ccip/onRamp/OnRampWithMessageTransformer.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IMessageTransformer} from "../interfaces/IMessageTransformer.sol";
+import {Internal} from "../libraries/Internal.sol";
+import {OnRamp} from "./OnRamp.sol";
+
+/// @notice OnRamp that uses a message transformer to transform messages from router
+contract OnRampWithMessageTransformer is OnRamp {
+  address internal s_messageTransformer;
+
+  error ZeroAddressNotAllowed();
+
+  constructor(
+    StaticConfig memory staticConfig,
+    DynamicConfig memory dynamicConfig,
+    DestChainConfigArgs[] memory destChainConfigs,
+    address messageTransformerAddr
+  ) OnRamp(staticConfig, dynamicConfig, destChainConfigs) {
+    if (address(messageTransformerAddr) == address(0)) {
+      revert ZeroAddressNotAllowed();
```

Comment:
Does it make sense to have this check in the constructor but not the setMessageTransformer function? I think the other code assumes it's always non-zero. Same for in the OffRamp

Reply:
yeah makes sense. added